File: Python/581A: Vasya the Hipster.py
# Read a and b
a, b = map(int, input().split())

# Fashionable days: need one of each color
different_days = min(a, b)

# Remaining socks of the same color
remaining = abs(a - b)

# Same-color days: need two of the same color
same_days = remaining // 2

print(f"{different_days} {same_days}")

File: CPP/581A: Vasya the Hipster.cpp
#include <iostream>
#include <algorithm> // for min
#include <cmath>     // for abs

using namespace std;

int main() {
    int a, b;
    cin >> a >> b;

    // The number of days he can wear different colors
    int different_days = min(a, b);
    
    // The number of socks left over
    int remaining = abs(a - b);
    
    // Two socks make one pair for the same-color days
    int same_days = remaining / 2;

    cout << different_days << " " << same_days << endl;

    return 0;
}
